src/components/webcams/JanusStreamer.vue

```diff
@@ -0,0 +1,153 @@
+<template>
+    <div>
+        <video
+            v-show="status === 'started'"
+            ref="stream"
+            class="webcamStream"
+            :style="webcamStyle"
+            autoplay
+            muted
+            playsinline />
+        <v-row v-if="status !== 'started'">
+            <v-col class="_webcam_webrtc_output text-center d-flex flex-column justify-center align-center">
+                <v-progress-circular v-if="status === 'connecting'" indeterminate color="primary" class="mb-3" />
+                <span class="mt-3">{{ status }}</span>
+            </v-col>
+        </v-row>
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Mixins, Prop, Ref, Watch } from 'vue-property-decorator'
+import { JanusJs, JanusSession, JanusStreamingPlugin } from 'typed_janus_js'
+import BaseMixin from '@/components/mixins/base'
+import { ConstructorOptions } from 'typed_janus_js/src/interfaces/janus'
+
+@Component
+export default class JanusStreamer extends Mixins(BaseMixin) {
+    private janusClient: JanusJs | null = null
+    private session: JanusSession | null = null
+    private handle: JanusStreamingPlugin | null = null
+    private useStun = false
+    private pc: RTCPeerConnection | null = null
+    private aspectRatio: null | number = null
+    private status: string = 'connecting'
+
+    @Prop({ required: true })
+    camSettings: any
+
+    @Prop({ default: null }) declare readonly printerUrl: string | null
+
+    @Ref() declare stream: HTMLVideoElement
+
+    get url() {
+        const baseUrl = this.camSettings.urlStream
+        let url = new URL(baseUrl, this.printerUrl === null ? this.hostUrl.toString() : this.printerUrl)
+        url.port = this.hostPort.toString()
+        url.protocol = this.printerUrl?.startsWith('https') ? 'wss' : 'ws'
+
+        if (baseUrl.startsWith('ws')) url = new URL(baseUrl)
+
+        return url
+    }
+
+    get streamId() {
+        return /stream=(\d+)/.exec(this.camSettings.urlStream)?.pop()
```

Comment:
i think it would be better not to use a "wrong" get variable here, just write the id after the /. like the following example:

`ws://123.123.123.123/streamid`